udf-compiler/src/main/scala/com/nvidia/spark/udf/Plugin.scala

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright (c) 2019-2020, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.nvidia.spark.udf
+
+import org.apache.spark.internal.Logging
+import org.apache.spark.sql.SparkSessionExtensions
+import org.apache.spark.sql.catalyst.expressions.{Alias, Expression, NamedExpression, ScalaUDF}
+import org.apache.spark.sql.catalyst.plans.logical.{LogicalPlan, Project}
+import org.apache.spark.sql.catalyst.rules.Rule
+
+import com.nvidia.spark.rapids.RapidsConf
+
+class Plugin extends Function1[SparkSessionExtensions, Unit] with Logging {
+  override def apply(extensions: SparkSessionExtensions): Unit = {
+    logWarning("Installing extensions to enable rapids UDF compiler support." +
+      s" To disable rapids UDF compiler support set [a setting] to false")
```

Comment:
For current stage, I removed the second sentence and kept only the first for logging.
My point is: this udf extension works totally for the sql-plugin. It should always goes with sql-plugin. If we have no cases for "sql-plugin enabled + udf-plugin disabled", we can always turn it on.

Reply:
I disagree that it will always work with the sql-plugin.  If I do have java code that does `x/y` That will get translated into a catalyst operation.  But if `x` and `y` are both longs and `y` happens to be `0`, the catalyst operation will return a NULL, but a scala UDF would fail the entire job with a `java.lang.ArithmeticException: / by zero`.  The closer we can get to making them semantically the same the better, but I don't see us matching it 100% any time soon.  So I really do want the ability to control this separately from the sql plugin.

Reply:
Would it still require users to specify `--conf spark.sql.extensions="com.nvidia.spark.SQLPlugin,com.nvidia.spark.udf.Plugin"` to enable the udf compiler, or just `--conf spark.sql.extensions="com.nvidia.spark.SQLPlugin"`?

Reply:
Nowadays we are using `spark.plugins=com.nvidia.spark.SQLPlugin`, and this can mutate the extensions config on initialization. So instead we can have another config to toggle it on and off, and later on it could be enabled by default.

Reply:
I personally think it makes sense to have it as an opt-in feature, i.e., have another config to toggle it, for the first release.

Reply:
I've added a config  `spark.rapids.sql.udfCompiler.enabled` to do the control. Please help review the related description in both codes and markdown docs.
And for supported udf operands, I plan to add them along with the detailed compiler implementation PR.